### PR TITLE
Include Attach dependencies

### LINKED ISSUE #54: Published pom doesn't include Attach dependencies
Attach (storage/util) is used as compileOnly dependencies, so these are not included in the published pom. 
This works only if the project that uses Maps already includes Attach, otherwise it will fail.

### PR DESCRIPTION
**Pull request by jperedadnr**
Fixes #54

### CODE REVIEW

File: maps/src/main/java/com/gluonhq/impl/maps/tile/osm/CachedOsmTileRetriever.java [hunk 1]
```diff
@@ -54,11 +56,15 @@
...
-
-            cacheRoot = new File(storageRoot, ".gluonmaps");
+            Optional<File> storageRoot = StorageService.create()
+                    .flatMap(StorageService::getPrivateStorage);
+            if (storageRoot.isEmpty()) {
+                cacheRoot = Files.createTempDirectory(".gluonmaps").toFile();
+                logger.info("StorageService implementation not found.\n" +
+                        "Tiles will be cached at a temporary directory: " + cacheRoot);
+            } else {
+                cacheRoot = new File(storageRoot.get(), ".gluonmaps");
```
Comment (abhinayagarwal) [inline maps/src/main/java/com/gluonhq/impl/maps/tile/osm/CachedOsmTileRetriever.java:66]: How about `~/.gluon/maps` ?

File: mavenPublish.gradle [hunk 2]
```diff
@@ -26,94 +26,75 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+apply plugin: 'java-library'
```
Comment (abhinayagarwal) [inline mavenPublish.gradle:29]: This could be moved to the parent `build.gradle`

File: maps/build.gradle [hunk 3]
```diff
@@ -27,10 +27,10 @@
...
 
 plugins {
-    id 'com.github.ben-manes.versions' version '0.20.0'
-    id 'com.github.hierynomus.license' version '0.15.0'
-    id 'net.nemerosa.versioning' version '2.8.2'
-    id 'maven'
+    id 'com.github.ben-manes.versions' version '0.39.0'
+    id 'com.github.hierynomus.license' version '0.16.1'
+    id 'net.nemerosa.versioning' version '2.14.0'
+    id 'maven-publish'
```
Comment (abhinayagarwal) [inline maps/build.gradle:33]: This could be removed as it is already a part of `mavenPublish.gradle`

File: mavenPublish.gradle [hunk 4]
```diff
@@ -26,94 +26,75 @@
...
+                    developer {
+                        name = 'Johan Vos'
+                        email = 'johan.vos@gluonhq.com'
+                        organization = 'Gluon'
+                        organizationUrl = 'http://gluonhq.com'
+                    }
+                }
+                licenses {
+                    license {
+                        name = 'GNU General Public License (GPL)'
```
Comment (abhinayagarwal) [inline mavenPublish.gradle:64]: Shouldn't this be GPL+CE ?

**Comment by abhinayagarwal** [comment]
Mobile sample `build.gradle` has a repository link with `http` instead of `https`. The project fails to build in IntelliJ because of this.

File: maps/src/main/java/com/gluonhq/impl/maps/tile/osm/CachedOsmTileRetriever.java (hunk 1 above)
Comment (jperedadnr) [reply maps/src/main/java/com/gluonhq/impl/maps/tile/osm/CachedOsmTileRetriever.java:66]: `.gluonmaps` is all time default location for Maps, if we want to change it, it's worth its own issue and PR, isn't it?

Comment (abhinayagarwal) [reply maps/src/main/java/com/gluonhq/impl/maps/tile/osm/CachedOsmTileRetriever.java:66]: ok

**Comment by jperedadnr** [comment]
Fixed (I was planning to address that issue within #55)

File: mavenPublish.gradle (hunk 2 above)
Comment (jperedadnr) [reply mavenPublish.gradle:29]: no, there is no parent build for this one.

Comment (abhinayagarwal) [reply mavenPublish.gradle:29]: With parent I meant the section in `build.gradle` where we have `apply from: rootProject.file('gradle/mavenPublish.gradle')`

Comment (jperedadnr) [reply mavenPublish.gradle]: ok, done

**Review by abhinayagarwal — APPROVED** [review]

**Pull request merged**